quic/s2n-quic-core/src/recovery/congestion_controller/fuzz_target.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    packet::number::PacketNumberSpace,
+    path::MINIMUM_MTU,
+    random,
+    recovery::{
+        bbr::BbrCongestionController, CongestionController, CubicCongestionController, RttEstimator,
+    },
+    time::{testing::Clock, Clock as _, Timestamp},
+};
+use bolero::{check, generator::*};
+use core::time::Duration;
+use std::collections::VecDeque;
+
+struct SentPacketInfo<PacketInfo> {
+    sent_bytes: u16,
+    time_sent: Timestamp,
+    cc_packet_info: PacketInfo,
+}
+
+#[derive(Debug, TypeGenerator)]
+enum Operation {
+    IncrementTime {
+        /// The milli-second value by which to increase the timestamp
+        millis: u16,
+    },
+    PacketSent {
+        #[generator(1..=255)]
+        count: u8,
+        #[generator(1200..=9000)]
+        bytes_sent: u16,
+        app_limited: Option<bool>,
+    },
+    RttUpdated {
+        #[generator(1..=2000)]
+        millis: u64,
+    },
+    AckReceived {
+        #[generator(1..=255)]
+        count: u8,
+    },
+    PacketLost,
```

Comment:
It might be a good idea to have these pick a random index in the `send_packets` queue, rather than popping in order.

Reply:
I ended up having the `AckReceived` event start at a random index and then remove the next `count` packets from that index